staging/src/k8s.io/component-helpers/scheduling/corev1/nodeaffinity/nodeaffinity.go

```diff
@@ -239,11 +223,10 @@ func nodeSelectorRequirementsAsSelector(nsm []v1.NodeSelectorRequirement, path *
 			errs = append(errs, field.NotSupported(p.Child("operator"), expr.Operator, nil))
 			continue
 		}
-		r, err := labels.NewRequirement(expr.Key, op, expr.Values)
+		r, err := labels.NewRequirement(expr.Key, op, expr.Values, field.WithPath(p))
 		if err != nil {
-			// TODO(#96167): Use error as returned by labels.NewRequirement once it
-			// is a field.Error.
-			errs = append(errs, field.Invalid(p, expr, err.Error()))
+			fieldErr := err.(*field.Error)
```

Comment:
No need to do this, as it is guaranteed.

However, since the plan is to return all errors, you have to call `errors.Flatten(errors.NewAggregate(errs))` at the end.

See https://github.com/kubernetes/kubernetes/blob/80be1d6c7217b99fb6ff083a190b44af53a628fc/pkg/scheduler/apis/config/validation/validation_pluginargs.go#L299 as an example

Reply:
I made some changes to make sure `nodeSelectorRequirementsAsSelector` still returns field.ErrorList required by `parsedTerm.parseErrs`

see https://github.com/kubernetes/kubernetes/pull/97538#issuecomment-752837158